python/tskit/trees.py

```diff
@@ -4186,6 +4186,9 @@ def at(self, position, **kwargs):
         Returns the tree covering the specified genomic location. The returned tree
         will have ``tree.interval.left`` <= ``position`` < ``tree.interval.right``.
         See also :meth:`Tree.seek`.
+        Note that as tree sequences are stored tree-by-tree this method has to pass
```

Comment:
Maybe a warning block would stand out a bit more? Something like "The current implementation of this operation is linear in the number of trees, so may be inefficient for large tree sequences. See `<this issue <https://github.com/tskit-dev/tskit/issues/684>`_ for more information."

It's nothing fundamental, we just haven't implemented the necessary indexing to do it.